peeringmod.py

```diff
@@ -0,0 +1,233 @@
+import json
+import os
+import subprocess
+import sys
+from collections import defaultdict
+from dataclasses import asdict, dataclass
+from ipaddress import IPv4Network, IPv6Network
+from typing import List
+from wsgiref import headers
+
+import jinja2
+import jsonschema
+import requests
+
+
+@dataclass
+class AnnouncementGenerator:
+    prefix: IPv4Network or IPv6Network
+    muxes: List[str]
+    details: List[dict]
+
+    def __init__(self, prefix, muxes, **kwargs):
+        self.prefix = prefix
+        self.muxes = muxes
+        self.details = list()
+
+        self.add_details(muxes, **kwargs)
+
+    def add_details(self, muxes, **kwargs):
+        dictdetails = dict()
+
+        dictdetails["muxes"] = muxes
+        for key, value in kwargs.items():
+            dictdetails[key] = value
+
+        self.details.append(dictdetails)
+
+    def create_announcement(self) -> defaultdict:
+        announcement = defaultdict(dict)
+        announcement[self.prefix]["announce"] = self.details
+        return announcement
+
+    def create_withdrawal(self) -> defaultdict:
+        announcement = defaultdict(dict)
+        announcement[self.prefix]["withdraw"] = self.muxes
+        return announcement
+
+    def validate(self, schema_fn):
+        with open(schema_fn, encoding="utf8") as fd:
+            schema = json.load(fd)
+        announce = self.create_announcement()
+        jsonschema.validate(announce, schema)
+
+    def save_as_json(self, filename: str, dir_path: str = "", withdraw: bool = False):
+        path = os.path.join(dir_path, f"{filename}.json")
+        if withdraw:
+            obj = json.dumps(self.create_withdrawal(), indent=2)
+        else:
+            obj = json.dumps(self.create_announcement(), indent=2)
+        with open(path, "w") as f:
+            f.write(obj)
+
+    def __repr__(self) -> str:
+        return f"{self.prefix}: {self.details}"
+
+
+@dataclass
+class ExperimentGenerator:
+    rounds_list: list
+
+    def __init__(self):
```

Comment:
When overwriting `__init__`, we should not use `dataclasses`.  The main idea behind `dataclasses` is that it will do the initialization for us (here we're doing initialization by hand with `__init__` and `add_round`).  I think we want to either remove `__init__` and `add_round` *or* remove `dataclass`; having both does not make sense.